Vendor update

AutomatedTester opened the pull request:
This updates tools/third_party/pytest and tools/third_party/py to their latest.

This adds `attrs`, `funcsigs`, and `pluggy` as dependecies which are needed in pytest in a `third_party` directory.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/w3c/web-platform-tests/8535)
<!-- Reviewable:end -->

AutomatedTester commented:
I am happy to redo this if we dont want the history from subtrees.

w3c-bots commented:
# Build BROKEN

Started: 2017-12-01 14:08:30
Finished: 2017-12-01 14:21:46


<h2>Failing Jobs</h2>
<ul>

<li>tools_unittest in py27</li>

<li>tools_unittest in py36</li>

<li>tools_unittest in pypy</li>

</ul>




View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/21348)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/310050550)

gsnedders commented:
@AutomatedTester we've previously avoided history from subtrees

jgraham commented:
Yeah, I think we should do this with `--squash`

jgraham commented:
Also if you have to vendor in a bunch of deps, can we create `tools/thirdparty` and we can eventually figure out how to move most stuff to there.

AutomatedTester commented:
ok, will update accordingly

AutomatedTester committed:
Merge commit '97bf1b88861c87075dbb73bcc604a196f828c077' as 'tools/third_party/pluggy'

AutomatedTester committed:
Squashed 'tools/third_party/pluggy/' content from commit bea8b75738

git-subtree-dir: tools/third_party/pluggy
git-subtree-split: bea8b7573868df8bfa65ee6d785d88e418b92c87

AutomatedTester committed:
Merge commit 'b31c42ef66498176535de880a38bd9521069c18c' as 'tools/third_party/attrs'

AutomatedTester committed:
Squashed 'tools/third_party/attrs/' content from commit defa6b6f40

git-subtree-dir: tools/third_party/attrs
git-subtree-split: defa6b6f40b88d878aa480d540ab60c46dec3fea

AutomatedTester committed:
Squashed 'tools/third_party/funcsigs/' content from commit db7f0afe3e

git-subtree-dir: tools/third_party/funcsigs
git-subtree-split: db7f0afe3ed219d4f27073226e75c77073c9d137

AutomatedTester committed:
Merge commit '2e7047fd07a870b725ef5b36e7f86bd65e45cc29' as 'tools/third_party/funcsigs'

AutomatedTester committed:
Squashed 'tools/third_party/pytest/' content from commit 2e8b0a83fe

git-subtree-dir: tools/third_party/pytest
git-subtree-split: 2e8b0a83fe467a3a8365b6e661e44e54d3a00b03

AutomatedTester committed:
Merge commit '857f915013a1f07cceecddf71a92cb0da734233b' as 'tools/third_party/pytest'

AutomatedTester committed:
Squashed 'tools/third_party/py/' content from commit 35b742df9e

git-subtree-dir: tools/third_party/py
git-subtree-split: 35b742df9efc8778c15e4d716d7d31b1c2adbf8a

AutomatedTester committed:
Merge commit '194be86e6f8c432d4bc29df5050eda65e6ddcf23' as 'tools/third_party/py'

AutomatedTester committed:
Remove py and pytest directories/subtrees in favour of new third_party
subtrees

AutomatedTester committed:
Update paths used for vendored libraries in third_party

AutomatedTester commented:
@jgraham @gsnedders I've updated with squashed subtrees and have moved pytest and it's dependencies into `third_party`

AutomatedTester committed:
Use the test file and test name only when comparing tests against the
metadata

pytest has updated their reporting to show the path to the tests in
results. We now need to just take the test_file::test_method instead
of path/to/the/test_file::test_method

jgraham commented on tools/wptrunner/wptrunner/executors/pytestrunner/runner.py:
```diff
@@ -117,7 +117,7 @@ def record_skip(self, report):
         if stack is not None:
             stack = str(stack)
-        new_result = (test, status, message, stack)
+        new_result = (test.split("/")[-1], status, message, stack)
```
This should perhaps be `os.path.sep`. But why do we put the test name in the subtest name at all? That just makes things fragile under renames. Can we split on `::` or whatever the separator between the path and the function name is?

AutomatedTester replied:
Agree on the `os.path.sep`.

As for the names of this, we currently use the file name is the metadata, https://dxr.mozilla.org/mozilla-central/source/testing/web-platform/meta/webdriver/tests/sessions/get_timeouts.py.ini#2

Changing this works might mean we have to redo all the metadata. Is that something we dont mind about (Will this break other vendors WPT?)

AutomatedTester commented:
Try run on Mozilla infra with these code changes https://treeherder.mozilla.org/#/jobs?repo=try&revision=66c48462a122

jgraham commented:
It might break all the metadata, but only subtest names and I think it will make things better in the long run. We get semi-frequent breakage when files are moved around as it is, and we can prevent that.

gsnedders commented:
You need to update `tools/tox.ini` to have the right set of exclude paths, I think, to get Travis CI passing.

jgraham commented:
Seems like the `pywebsocket` changes are badness.

AutomatedTester commented:
@jgraham I havent updated the `pywebsocket` dependencies as it doesnt look like Google has pushed anything to that repo in years.

AutomatedTester commented:
@gsnedders updated the files used for testing. We need docs on how to run tests properly for `tools`

AutomatedTester commented:
@jgraham the only thing that I can find from my try run is https://github.com/w3c/web-platform-tests/commit/d8ca91efb791173048ea51592625915e690c7a2e which looks quite innocuous so guidance here would be appreciated

jgraham committed:
Set computed default values when loading server config in wptrunner. (#8589)

The recent changes to configuration in wptserve broke websockets tests
running in wptrunner because the server was being passed None as the
ws_doc_root, so it errored on startup. We can force this value to be
set by running set_computed_defaults on the config.

A longer term fix would remove the separate configuration for the
server in environment.py so that it can run through the same codepath
as in serve.py.

AutomatedTester commented:
Latest [try run](https://treeherder.mozilla.org/#/jobs?repo=try&revision=cc5829c3596c) with this PR and cherry pick of https://github.com/w3c/web-platform-tests/commit/775e4b338a908fac840d118c0350d53f9da558f9

AutomatedTester commented:
current [try](https://treeherder.mozilla.org/#/jobs?repo=try&revision=5687db1144b5) looks good for a merge

AutomatedTester commented:
current [try](https://treeherder.mozilla.org/#/jobs?repo=try&revision=cbeb2fc81406) looks good for a merge
